src/world/script_api/7E0E80.c

```diff
@@ -1,16 +1,131 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "world/script_api/7E0E80", func_80280000);
+extern s32 MessagePlural;
+extern s32 MessageSingular;
+
+extern s32 D_80283E80_7E4D00;
+extern s32 D_80283EB0_7E4D30;
+extern s32 D_80283EE0_7E4D60;
+extern s32 D_80283F2C_7E4DAC;
+extern s32 D_80284034_7E4EB4;
+extern s32 D_80286520;
+extern s32 D_80286524;
+
+s32 shop_owner_begin_speech(s32 messageIndex) {
+    Shop* shop = gGameStatusPtr->mapShop;
+    s32 shopStringID = shop->owner->shopStringIDs[messageIndex];
+    ScriptInstance* script = start_script(&D_80283E80_7E4D00, 1, 0);
+
+    script->varTable[0] = shopStringID;
+    script->varTable[1] = shop->owner->npcID;
+    script->varTable[2] = shop->owner->talkAnim;
+    script->varTable[3] = shop->owner->idleAnim;
+
+    return script->id;
+}
+
+s32 shop_owner_buy_dialog(s32 messageIndex, s32 itemName, s32 coinCost, s32 bpCost) {
+    Shop* shop = gGameStatusPtr->mapShop;
+    s32 shopStringID = shop->owner->shopStringIDs[messageIndex];
+    ScriptInstance *script;
+    s32 phi_a0;
+
+    set_message_string(itemName, 0);
+    set_message_value(coinCost, 1);
+
+    if (bpCost > 0) {
+        set_message_value(bpCost, 2);
+    } else {
+        if (coinCost == 1) {
+            phi_a0 = &MessageSingular;
+        } else {
+            phi_a0 = &MessagePlural;
+        }
+        set_message_string(phi_a0, 2);
```

Comment:
can probably turn this into
```c
        if (coinCost == 1) {
            set_message_string(&MessageSingular, 2);
        } else {
            set_message_string(&MessagePlural, 2);
        }
```

Reply:
I can't - I tried. And because I couldn't do it here, I didn't do it elsewhere. But I could do the other one I guess

Reply:
sad

Reply:
doesn't work